lib/utils/which.rb

```diff
@@ -0,0 +1,16 @@
+# encoding: utf-8
+
+# Cross-platform way of finding an executable in the $PATH.
+#
+#   which('ruby') #=> /usr/bin/ruby
+def which(cmd)
```

Comment:
It might be good to take an optional second argument, providing a search path and defaulting to ENV['PATH'].  Refs #450, attn @miah

Reply:
Taking a option second argument would definitely be helpful.